app/src/main/java/com/example/superimagesviewer/MosaicViewModel.java

```diff
@@ -0,0 +1,42 @@
+package com.example.superimagesviewer;
+
+import android.app.Application;
+import android.content.res.AssetManager;
+import android.graphics.drawable.Drawable;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MosaicViewModel extends AndroidViewModel {
+
+    private static final String IMAGES_FOLDER_NAME = "MyImages";
+    private final MutableLiveData<List<Drawable>> mosaicsList =  new MutableLiveData<>();
+    private final List<Drawable> drawables = new ArrayList<>();
+
+    public MosaicViewModel(@NonNull Application application) throws IOException {
+        super(application);
+        loadImagesFromAssets();
```

Comment:
Please avoid usage of any heavy operations inside constructor code. In constructor we can only init objects but never load something from file system or network

Reply:
For sure this rule only applies for main thread operations. If you start some background operation in constructor, then we are fine

Reply:
done